app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt

```diff
@@ -0,0 +1,61 @@
+package com.arygm.quickfix.model.locations
+
+import java.io.IOException
+import okhttp3.*
+import org.json.JSONArray
+
+class NominatimLocationRepository(val client: OkHttpClient) : LocationRepository {
+  override fun search(
+      query: String,
+      onSuccess: (List<Location>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val url = "https://nominatim.openstreetmap.org/search?q=${query}&format=json"
+
+    val request =
+        Request.Builder()
+            .url(url)
+            .header(
+                "User-Agent", "QuickFix/1.0 (daferyassine52@gmail.com)") // Set a proper User-Agent
+            .header("Referer", "https://QuickFix.com") // Optionally add a Referer
```

Comment:
DNS names are all lowercase:
```suggestion
            .header("Referer", "https://quickfix.com") // Optionally add a Referer
```

Reply:
Thank you for pointing this out! I have updated the DNS name to lowercase as suggested  Please let me know if there’s anything else to adjust.